frontend/packages/operator-lifecycle-manager/src/components/operand/DEPRECATED_operand-form.tsx

```diff
@@ -643,14 +643,17 @@ export const DEPRECATED_CreateOperandForm: React.FC<OperandFormProps> = ({
       groupName,
       fieldLists: _.reduce(
         fieldsInGroup,
-        (fieldListsAccumulator, field) => {
+        (fieldListsAccumulator: OperandField[][], field) => {
           const { index, regexMatch } = parseArrayPath(field.path);
           if (regexMatch) {
-            fieldListsAccumulator[index] = [...(fieldListsAccumulator[index] || []), field];
+            fieldListsAccumulator[index || 0] = [
+              ...(fieldListsAccumulator[index || 0] || []),
+              field,
+            ];
           }
           return fieldListsAccumulator;
         },
-        [],
+        [] as OperandField[][],
```

Comment:
IIRC, lodash reduce supports a generic, which should simplify this a bit:
```suggestion
fieldLists: _.reduce<OperandField[][]>(
fieldsInGroup,
        (fieldListsAccumulator, field) => {
const { index, regexMatch } = parseArrayPath(field.path);
if (regexMatch) {
            fieldListsAccumulator[index || 0] = [
              ...(fieldListsAccumulator[index || 0] || []),
              field,
            ];
}
return fieldListsAccumulator;
},
        [],
```